### PR TITLE
feat: if improvements

### PR DESCRIPTION
**Pull request by kiliantyler**
### Prerequisites

- [x] I have read and understood the [contributing guide][CONTRIBUTING.md].
- [x] The commit message follows the [conventional commits][cc] guidelines.
- [x] Tests for the changes have been added (for bug fixes / features).
- [x] Docs have been added/updated (for bug fixes / features).

## Main new features
- `If` now supports both a string and a list
- [slim-sprig][slim-sprig] added to templates
- new runtime variables (.User, .ConfigDir, .CacheDir, .Hostname)

### Ifs
This allows `If` to be a string as it currently is:
```bash
alias:
  - name: hello
    value: world
    if: eq .Shell "zsh"
```

But also allows it to be a list:
```bash
alias:
  - name: hello
    value: world
    if:
      - eq .Shell "zsh"
      - match .User "root"
```

### slim-sprig
This incorporates [slim-sprig][slim-sprig] template functions into templates. The slim version was chosen since most of the features in the full version that are missing are expensive and unneeded.

### New runtime variables
- `.User` -- The username of the current user
- `.Hostname` -- The current machine's hostname
- `.ConfigDir` -- The current user's config directory based on OS
  - Unix: `$XDG_CONFIG_HOME` (or `$HOME/.config` if empty)
  - MacOS: `$HOME/Library/Application Support`
  - Windows: `%AppData%`
- `.CacheDir` -- The current user's cache directory based on OS
  - Unix: `$XDG_CACHE_HOME` (or `$HOME/.cache` if empty)
  - MacOS: `$HOME/Library/Caches`
  - Windows: `%LocalAppData%`
 
## Additional changes
- Refactored the various rendering switches to be in one place, allowing for new shells to be added easier. Also added interfaces for the various "types" of exports to conform to `Renderer`
- Added install instructions for the [zinit][zinit] `zsh` plugin manager

[CONTRIBUTING.md]: https://github.com/JanDeDobbeleer/aliae/blob/main/CONTRIBUTING.md
[cc]: https://www.conventionalcommits.org/en/v1.0.0/#summary
[slim-sprig]: https://go-task.github.io/slim-sprig/
[zinit]: https://github.com/zdharma-continuum/zinit

### CODE REVIEW

File: src/context/runtime.go [hunk 1]
```diff
@@ -10,30 +10,71 @@ import (
...
+	}
+	return ""
+}
+
+func Hostname() string {
+	if Current != nil {
+		return Current.Hostname
+	}
+	hostname, err := os.Hostname()
+	if err == nil && len(hostname) > 0 {
```
Comment (JanDeDobbeleer) [inline src/context/runtime.go:65]: `len(hostname) > 0` is obsolete as we return `""` anyways on error. So I don't see a change in behaviour.

File: src/shell/if.go [hunk 2]
```diff
@@ -22,3 +22,42 @@ func (i If) Ignore() bool {
...
+		str, ok := value.(string)
+		if !ok {
+			continue
+		}
+		strs[index] = str
+	}
+	return strs
+}
+
+func checkIf(i any) bool {
```
Comment (JanDeDobbeleer) [inline src/shell/if.go:48]: If `If` is a custom type instead of `interface{}`, this can also be a function on that type. Will make this cleaner.

**Review by JanDeDobbeleer — CHANGES_REQUESTED** [review]

**Comment by JanDeDobbeleer** [comment]
@kiliantyler I was still going to look at this, I know it took ages but was on my list. I understand if that was a bit too much an ask on your end, but it's not forgotten. Oh-my-posh just took up a lot of my time the past months.

**Comment by kiliantyler** [comment]
Totally fine! I actually deleted my fork because I was just cleaning up some repos and I needed to redo it. Totally forgot about this PR! Sorry!

**Comment by JanDeDobbeleer** [comment]
@kiliantyler ok cool, if you want to pick this up again, I'll have a look over the weekend.

**Pull request closed without merging**